packages/webapp/src/containers/Finances/CustomExpenseType/EditSimpleCustomExpense.jsx

```diff
@@ -0,0 +1,70 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import PureSimpleCustomType from '../../../components/Forms/SimpleCustomType';
+import { HookFormPersistProvider } from '../../hooks/useHookFormPersist/HookFormPersistProvider';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+import { updateCustomExpenseType } from '../actions';
+import { expenseTypeByIdSelector, expenseTypeSelector } from '../selectors';
+import { CUSTOM_EXPENSE_NAME } from './constants';
+
+function EditCustomExpense({ history, match }) {
+  const expense_type_id = match.params.expense_type_id;
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const onGoBackPath = `/readonly_custom_expense/${expense_type_id}`;
+  const persistedPaths = [onGoBackPath];
+  const selectedCustomExpenseType = useSelector(expenseTypeByIdSelector(expense_type_id));
+  const expenseTypes = useSelector(expenseTypeSelector);
+  const { expense_name } = selectedCustomExpenseType;
+
+  const handleGoBack = () => {
+    history.back();
+  };
+
+  const onSubmit = (payload) => {
+    dispatch(updateCustomExpenseType(payload, expense_type_id));
+  };
+
+  const validateUniqueTypeName = (value) => {
+    const expenseNameExists = expenseTypes.some((type) => {
+      return type.expense_name === value;
+    });
+
+    if (expenseNameExists) {
+      return t('EXPENSE.ADD_EXPENSE.DUPLICATE_NAME');
+    }
+    return true;
+  };
```

Comment:
Is this exactly the same as the one in the creation component??
I'm just wondering why you didn't make it reusable...

Reply:
No reason haha yes I should make it generic.